pyccel/ast/builtins.py

```diff
@@ -1485,10 +1485,11 @@ def __init__(self, *x):
 
         if isinstance(x, (list, tuple)):
             x = PythonTuple(*x)
-        elif not isinstance(x, (PythonTuple, PythonList)):
+        elif not (isinstance(x, (PythonTuple, PythonList))
+                  or isinstance(x, Variable) and isinstance(x.class_type, HomogeneousContainerType)):
```

Comment:
I can never remember which takes precedence between `or` and `and`. Maybe add some parentheses for clarity. Same remark on line 1525